scripts/azure-pipelines/android/azure-pipelines.yml

```
@@ -0,0 +1,108 @@
+    default: 'x64-android'
+  - name: jobName
+    type: string
+    default: 'x64_android'
+  - name: poolName
+    type: string
+  - name: tripletPattern
+    displayName: 'Enable the triplets which contain this substring'
+    type: string
+    default: ''
+
+jobs:
+- job: ${{ parameters.jobName }}
+  condition: and(succeeded(), contains('^${{ parameters.triplet }}$', '${{ parameters.tripletPattern }}'))
+  pool:
+    name: ${{ parameters.poolName }}
+  workspace:
+    clean: resources
+  timeoutInMinutes: 1440 # 1 day
+  cancelTimeoutInMinutes: 1
+  variables:
+  - name: WORKING_ROOT
+    value: /mnt/vcpkg-ci
+  - name: VCPKG_DOWNLOADS
+    value: /mnt/vcpkg-ci/downloads
+  - group: vcpkg-asset-caching-credentials
+  - name: X_VCPKG_ASSET_SOURCES
+    value: "x-azurl,$(root-url-ea),$(sas-ea),readwrite"
+  - group: vcpkg-binary-caching-credentials
+  - name: X_VCPKG_BINARY_SOURCE_STUB
+    value: "x-azblob,$(root-bin-url-ea),$(sas-bin-ea)"
+
+  steps:
+    # Note: /mnt is the Azure machines' temporary disk.
+  - bash: |
+      sudo mkdir /home/agent -m=777
+      sudo chown `id -u` /home/agent
+      sudo mkdir ${{ variables.WORKING_ROOT }} -m=777
+      sudo rm -rf ${{ variables.WORKING_ROOT }}/failure-logs
+      sudo mkdir ${{ variables.WORKING_ROOT }}/failure-logs -m=777
+      sudo mkdir ${{ variables.VCPKG_DOWNLOADS }} -m=777
+      exit 0
+    displayName: 'Create working directories'
+  - bash: ./bootstrap-vcpkg.sh
+    displayName: 'Bootstrap vcpkg'
+    condition: eq('use default', '${{ parameters.vcpkgToolSha }}')
+  - bash: ./scripts/azure-pipelines/bootstrap-from-source.sh ${{ parameters.vcpkgToolSha }}
+    displayName: "Build vcpkg with CMake"
+    condition: ne('use default', '${{ parameters.vcpkgToolSha }}')
+  - bash: |
+      curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+      az login --identity
+      az acr login --name AndContainerRegistry
+      USER=$(id --user)
+      echo "##vso[task.setvariable variable=USER]$USER"
+    displayName: 'Set up managed identity'
+  - bash: |
+      docker stop $(docker ps -a -q)
+    displayName: 'Stop any running docker containers'
+  - bash: |
+      docker pull andcontainerregistry.azurecr.io/vcpkg-android:2023-03-23
+    displayName: 'Pull Android Docker Container'
+  - task: Bash@3
+    inputs:
+      targetType: inline
+      script: |
+        docker run --init -i -a stderr -a stdout --user $(USER) --mount type=bind,source=$(Build.Repository.LocalPath),target=/vcpkg --mount type=bind,source=$(WORKING_ROOT)/failure-logs,target=/vcpkg/failure-logs --mount type=bind,source=/mnt/vcpkg-ci,target=/mnt/vcpkg-ci --env X_VCPKG_ASSET_SOURCES="x-azurl,$(root-url-ea),$(sas-ea),readwrite" "andcontainerregistry.azurecr.io/vcpkg-android:2023-03-23" pwsh -File /vcpkg/scripts/azure-pipelines/test-modified-ports.ps1 -Triplet ${{ parameters.triplet }} -BuildReason $(Build.Reason) -BinarySourceStub "$(X_VCPKG_BINARY_SOURCE_STUB)" -WorkingRoot ${{ variables.WORKING_ROOT }}

```

Can this be broken into multiple lines somehow?

Maybe this (not sure if it works):
```suggestion
      script: |
        docker run --init -i \
          -a stderr \
          -a stdout \
          --user $(USER) \
          --mount type=bind,source=$(Build.Repository.LocalPath),target=/vcpkg \
          --mount type=bind,source=$(WORKING_ROOT)/failure-logs,target=/vcpkg/failure-logs \
          --mount type=bind,source=/mnt/vcpkg-ci,target=/mnt/vcpkg-ci \
          --env X_VCPKG_ASSET_SOURCES="x-azurl,$(root-url-ea),$(sas-ea),readwrite" \
          "andcontainerregistry.azurecr.io/vcpkg-android:2023-03-23" \
          pwsh \
            -File /vcpkg/scripts/azure-pipelines/test-modified-ports.ps1 \
            -Triplet ${{ parameters.triplet }} \
            -BuildReason $(Build.Reason) \
            -BinarySourceStub "$(X_VCPKG_BINARY_SOURCE_STUB)" \
            -WorkingRoot ${{ variables.WORKING_ROOT }}
```